File: Dev-Kumbh-Mela-Website-SETTribe-Backend/src/main/java/com/kumbh/service/HospitalService.java
package com.kumbh.service;

import com.kumbh.dto.HospitalDto;
import com.kumbh.entity.Hospital;
import com.kumbh.pagination.HospitalPagination;
import com.kumbh.repository.HospitalRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class HospitalService {

    @Autowired
    private HospitalRepository repository;

    @Autowired
    private HospitalPagination pagination;

    public List<Hospital> getAllHospitalsList() {
        return repository.findAll();
    }

    public Page<HospitalDto> getAllHospitals(String search, Integer page, Integer size, String sortBy, String direction) {
        return pagination.getPaginatedHospitals(search, page, size, sortBy, direction);
    }

    public Optional<Hospital> getHospitalById(Long id) {
        return repository.findById(id);
    }

    public Hospital createHospital(Hospital hospital) {
        return repository.save(hospital);
    }

    public Optional<Hospital> updateHospital(Long id, Hospital hospitalDetails) {
        return repository.findById(id).map(hospital -> {
            hospital.setName(hospitalDetails.getName());
            hospital.setAddress(hospitalDetails.getAddress());
            hospital.setLatitude(hospitalDetails.getLatitude());
            hospital.setLongitude(hospitalDetails.getLongitude());
            hospital.setContact(hospitalDetails.getContact());
            hospital.setBeds(hospitalDetails.getBeds());
            hospital.setStatus(hospitalDetails.getStatus());
            return repository.save(hospital);
        });
    }

    public boolean deleteHospital(Long id) {
        return repository.findById(id).map(hospital -> {
            hospital.setStatus("INACTIVE");
            repository.save(hospital);
            return true;
        }).orElse(false);
    }

    public boolean existsByName(String name) {
        return repository.existsByName(name);
    }

    public boolean existsByContact(String contact) {
        return repository.existsByContact(contact);
    }

    public boolean existsByLatitude(Double latitude) {
        return repository.existsByLatitude(latitude);
    }

    public boolean existsByLongitude(Double longitude) {
        return repository.existsByLongitude(longitude);
    }
}

File: Dev-Kumbh-Mela-Website-SETTribe-Frontend/src/api/services/hospitalService.js
import axiosInstance from '../axiosConfig';

export const hospitalService = {
    getAllHospitals: async (page = 0, size = 10, searchTerm = '') => {
        try {
            const params = new URLSearchParams();
            params.append('page', page);
            params.append('size', size);
            if (searchTerm) params.append('search', searchTerm);
            
            const response = await axiosInstance.get(`/api/hospitals?${params.toString()}`);
            return response;
        } catch (error) {
            throw error;
        }
    },

    getHospitalById: async (id) => {
        try {
            const response = await axiosInstance.get(`/api/hospitals/${id}`);
            return response;
        } catch (error) {
            throw error;
        }
    },

    createHospital: async (hospitalData) => {
        try {
            const response = await axiosInstance.post('/api/hospitals', hospitalData);
            return response;
        } catch (error) {
            throw error;
        }
    },

    updateHospital: async (id, hospitalData) => {
        try {
            const response = await axiosInstance.put(`/api/hospitals/${id}`, hospitalData);
            return response;
        } catch (error) {
            throw error;
        }
    },

    deleteHospital: async (id) => {
        try {
            const response = await axiosInstance.delete(`/api/hospitals/${id}`);
            return response;
        } catch (error) {
            throw error;
        }
    },

    checkName: async (name) => {
        try {
            const response = await axiosInstance.get(`/api/hospitals/check-name?name=${encodeURIComponent(name)}`);
            return response.data;
        } catch (error) {
            throw error;
        }
    },

    checkContact: async (contact) => {
        try {
            const response = await axiosInstance.get(`/api/hospitals/check-contact?contact=${encodeURIComponent(contact)}`);
            return response.data;
        } catch (error) {
            throw error;
        }
    },

    checkLatitude: async (latitude) => {
        try {
            const response = await axiosInstance.get(`/api/hospitals/check-latitude?latitude=${latitude}`);
            return response.data;
        } catch (error) {
            throw error;
        }
    },

    checkLongitude: async (longitude) => {
        try {
            const response = await axiosInstance.get(`/api/hospitals/check-longitude?longitude=${longitude}`);
            return response.data;
        } catch (error) {
            throw error;
        }
    }
};
